terraform/examples/module_depends_on/README.md

```diff
@@ -0,0 +1,98 @@
+# Example for the `module_depends_on` feature
+
+This example is discussed in more detail in our article
+[xxxxx](xxx).
+
+The code in main.tf defines two resources and two modules where
+one module depends on one of the resources.
+
+```
+module "has-no-dependency" {
+  ....
+}
+
+resource "null_resource" "is-no-dependency" {
+  ....
+}
+
+module "has-external-dependency" {
+  ....
+
+  module_depends_on = [
+    null_resource.is-external-dependency
+  ]
+}
+
+resource "null_resource" "is-external-dependency" {
+  ....
+}
+```
+
+## Running the example
+The example will only create `null` resources as provided by the [null provider]().
+No infrastructure is created anywhere and no costs will be generated.
+Running the example will us the [local backend]() by default to store the terraform state.
+
+### Cloning the repository
+```
+git clone https://github.com/mineiros-io/article-examples.git
+cd article-examples/terraform/examples/module_depends_on
+```
+
+### Initializing Terraform
+Run `terraform init` to initialize the example. The output should look like:
+```
+Initializing modules...
```

Comment:
You could also work with gifs for showing example output. [Terminalizer](https://github.com/faressoft/terminalizer) is a great tool for creating gifs.

Reply:
yeah for initialize this makes sense... but for the other examples it is hard to follow details in animations... so i would rather keep it aligned..